google/cloud/grpc_options.cc

```diff
@@ -22,6 +22,18 @@ namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 
+void ConfigureContext(grpc::ClientContext& context, Options const& opts) {
+  if (opts.has<GrpcSetupOption>()) {
+    opts.get<GrpcSetupOption>()(context);
+  }
+}
+
+void ConfigurePollContext(grpc::ClientContext& context, Options const& opts) {
+  if (opts.has<GrpcSetupPollOption>()) {
+    opts.get<GrpcSetupPollOption>()(context);
```

Comment:
Consider:

```cc
void F() {
  grpc::ClientContext context;
  std::function<void(grpc::ClientContext&)> crashme;
  google::cloud::internal::ConfigureContext(
    context, google::cloud::Options{}.set<google::cloud::GrpcSetupOption>(crashme);
}
```

Do you want to check that the `std::function<>` is non-empty or something?

Reply:
(I did not think about this). Umm, I could add a check... but we do not check that a custom retry policy is not `nullptr` before using it either. ([1](https://github.com/googleapis/google-cloud-cpp/blob/33ce1b369bb37e7a97326c06fdf7c61d0933bfb5/google/cloud/bigquery/internal/bigquery_read_option_defaults.cc#L57)) ([2](https://github.com/googleapis/google-cloud-cpp/blob/33ce1b369bb37e7a97326c06fdf7c61d0933bfb5/google/cloud/bigquery/bigquery_read_connection.cc#L71))

Should we think of passing an empty `std::function` as any different than passing a null pointer?

We could start checking for both....

Reply:
Whoops, I did not answer your question, I just said it back at you. 

My answer is a weak no, we do not want to check that it is not empty.

Reply:
SGTM.